docs/developing/styleguide.rst

```diff
@@ -4,31 +4,37 @@ Coding Style Guide
 
 Follow `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_ (the official style guide for Python).
 
+However, for Ion, we limit the lengths of lines to 150 characters, not 80 characters.
+
 Main points:
 
 - Indent using 4 spaces.
 - Use underscores in favor of camel case for all names except the names of classes.
-- Limit all lines to a maximum of 79 characters.
 - Limit the line length of docstrings or comments to 72 characters.
 - Separate top-level functions and class definitions with two blank lines.
 - Separate method definitions inside a class with a single blank line.
-- Use two spaces before inline comment and one space between the pound sign and comment/
+- Use two spaces before inline comment and one space between the pound sign and comment.
 - Use a plugin for your text editor to check for/remind you of PEP8 conventions.
-- Capitalize and punctuate comments and git commit messages properly.
+- Run the ``flake8`` linter periodically.
+- Capitalize and punctuate comments and Git commit messages properly.
 
 Imports
 =======
 
 - Group imports in the following order:
     #. Standard library imports
-    #. Imports from core Django
-    #. Related third-party imports
-    #. Local application or library specific imports, imports from Django apps
+    #. Third-party imports
+    #. Imports from Django
```

Comment:
Because we order our imports along the lines of:
```
import os

import redis

from django.urls import reverse
from django.utils import timezone

from .models import Model
```
which is standard lib -> third-party -> Django -> local imports

Reply:
A quick `grep` through the codebase shows that we're actually not very consistent one way or the other. There are places where we have Django imports before third-party imports, and there are places where it's the other way around.

Recently, I've been grouping Django imports and third-party imports together and ordering alphabetically. I think that if we split them up then we should do stdlib -> Django -> third-party -> local because in Django applications Django is almost like part of the standard library and thus (in my brain) should come first.

Reply:
Lets move this discussion over to a dedicated issue and not let this block a merge.

Reply:
Let's move this discussion to a separate issue (https://github.com/tjcsl/ion/issues/739) and not let this block a merge.